src/applications/MachineShopSimulator.java

```diff
@@ -21,171 +20,115 @@
-            }
-        } else {// task has just finished on machine[theMachine]
-                // schedule change-over time
-            lastJob = machine[theMachine].getActiveJob();
-            machine[theMachine].setActiveJob(null);
-            eList.setFinishTime(theMachine, timeNow
-                    + machine[theMachine].getChangeTime());
-        }
-
-        return lastJob;
-    }
-
     private void setMachineChangeOverTimes(SimulationSpecification specification) {
         for (int i = 1; i<=specification.getNumMachines(); ++i) {
             machine[i].setChangeTime(specification.getChangeOverTimes(i));
         }
     }
 
+    /**
+     * sets up the jobs for the task queue
+     * @param specification
+     */
     private void setUpJobs(SimulationSpecification specification) {
-        // input the jobs
         Job theJob;
         for (int i = 1; i <= specification.getNumJobs(); i++) {
             int tasks = specification.getJobSpecifications(i).getNumTasks();
-            int firstMachine = 0; // machine for first task
+            int firstMachine = 0;
 
-            // create the job
             theJob = new Job(i);
             for (int j = 1; j <= tasks; j++) {
                 int theMachine = specification.getJobSpecifications(i).getSpecificationsForTasks()[2*(j-1)+1];
                 int theTaskTime = specification.getJobSpecifications(i).getSpecificationsForTasks()[2*(j-1)+2];
                 if (j == 1)
-                    firstMachine = theMachine; // job's first machine
-                theJob.addTask(theMachine, theTaskTime); // add to
-            } // task queue
+                    firstMachine = theMachine;
+                theJob.addTask(theMachine, theTaskTime);
+            }
             machine[firstMachine].getJobQ().put(theJob);
         }
     }
 
     private void createEventAndMachineQueues(SimulationSpecification specification) {
-        // create event and machine queues
         eList = new EventList(specification.getNumMachines(), largeTime);
         machine = new Machine[specification.getNumMachines() + 1];
         for (int i = 1; i <= specification.getNumMachines(); i++)
             machine[i] = new Machine();
     }
 
-    /** load first jobs onto each machine
-     * @param specification*/
+    /**
+     * load first jobs onto each machine
+     * @param specification
+     */
     void startShop(SimulationSpecification specification) {
-        // Move this to startShop when ready
         numMachines = specification.getNumMachines();
         numJobs = specification.getNumJobs();
         createEventAndMachineQueues(specification);
-
-        // Move this to startShop when ready
         setMachineChangeOverTimes(specification);
-
-        // Move this to startShop when ready
         setUpJobs(specification);
 
         for (int p = 1; p <= numMachines; p++)
-            changeState(p);
+            Machine.changeState(p, machine[p], eList, timeNow);

```

Comment:
A similar comment would apply here:

```suggestion
            machine[p].changeState(p, eList, timeNow);
```